### ISSUE TITLE
Option to run a pulse/alert using each recipient's privileges

### ISSUE THREAD

**Issue by lindsay-stevens** [metabase/metabase#12091]
**Is your feature request related to a problem? Please describe.**

At the moment a pulse/alert sends the results that would be seen by the creator/owner of the pulse/alert. The results shown in questions may vary by user when sandboxing / row level security is in use.

The current behaviour is useful to set up updates which individual users might not otherwise be able to see, e.g. a summary of progress across all departments.

An option to run the pulse/alert as the recipient would be useful to set up updates which are personalised for each recipient e.g. a summary of progress auto-filtered for their department only. 

**Describe the solution you'd like**

When creating a pulse/alert, add a tick box that says something like "Run this pulse/alert's questions as if each user viewed them". If a recipient does not have a Metabase account, the current default of running the questions as the pulse/alert owner would be used (warn the user of that if they tick the box).

Since only admins will know for sure if sandboxing / RLS is in place, there should be a global setting to enable / disable this feature. Otherwise it might be confusing if everybody gets the same results and/or the backend might waste effort re-running questions only to get the same results every time.

**Describe alternatives you've considered**

I think the only alternative for this is for users to create their own copy of the pulse/alert. Additionally there no pulse/alert copy feature. So it is not particularly practical to expect users to manually create their own copy, especially if they aren't 100% invested in the contents.

**How important is this feature to you?**

I would say 3 out of 10. While it's not a main selling point of Metabase, it can be a source of disappointment once users start taking a deeper look at what they can do with Metabase.

**Additional context**

This may also affect / enhance query auditing in that it could produce more specific records of who has viewed what data. Maybe as part of this change, when a query is run as the pulse/alert owner, the recipient list should be added to the audit log record.

**Comment by snarfed** [comment]
yes! we'd dearly like this. closer to a 6 out of 10 priority for us. we're actively using sandboxed pulses, and initially we expected alerts would behave the same way, especially since the recipient picker lets you choose Metabase users, as opposed to raw email addresses, which somewhat implies that it's aware of and respects their sandboxing and other settings.

cc @shermanlaw
